fix: Log not showing in CLI

EdgE790 opened the pull request:
My implementation of #62  added two loggers where one use StreamHandler with stdout to separate output of different values to different pipes (stdout and stderr).
But it seems that stdout logger wasn't flushed properly. 
This PR should fix it by introducing FlushingStreamHandler which flushes after every log message.

This PR was not yet tested, because currently I can't test it, but I will do so as soon as possible (probably in 24 hours)

bogadana commented:
i told you this doesnt work properly, it prints twice

bogadana commented:
doing it like this works but doesnt seem like the best solution
```kt
    init {
        logger.useParentHandlers = false
        logger.handlers.forEach { logger.removeHandler(it) }
        logger.addHandler(FlushingStreamHandler(System.out, SimpleFormatter()))
    }
```

TheJeterLP commented:
Any update on this? This issue is really annoying

EdgE790 commented:
Ok, so I believe I found the problem.
FlushingLoggingHandler works fine, but DefaultCliLogger created twice.
Each creation invokes init block for it's instance, loggers inside DefaultCliLogger are reused if they have same name.
It leads to two handlers being added to the same logger, which results in duplicated output.

The simple (but not really good) solution will be to remove all handlers when we adding new one (I've checked it, the only handler that we add is ours).

This solution is bad because any configuration of the logger will be overridden, but we don't have any.

The proper solution may be to migrate to other logging frameworks that will have more flexible configuration.

EdgE790 commented:
Replaced removal of handlers with conditional addition of handler. Now it will be added only the first time.
Also changed name for the error handler to reuse first handler name.

EdgE790 commented:
Additionally it seems that 
`Logger.getLogger(MainCommand::javaClass.name)` is not working properly and returns constant string: "javaClass"
`Logger.getLogger(MainCommand::class.java.name)` seems to be working fine resulting in: "app.revanced.cli.command.MainCommand"

oSumAtrIX commented:
Is this ready to get merged?

EdgE790 commented:
Yep, working fine for me and all of the mentioned fixes are already in the PR
